contrib/cpp-dns-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+set(LIBRARY_DIR "${ClickHouse_SOURCE_DIR}/contrib/cpp-dns")
+
+set(SRCS "${LIBRARY_DIR}/DNSResolver.cpp")
+
+set(HDRS "${LIBRARY_DIR}/DNSResolver.hpp")
+
+add_library(_cpp-dns ${SRCS} ${HDRS})
```

Comment:
I am pretty sure we don't need to include `${HDRS}` into the library. Below `target_include_directories` call makes the header (or more precise: the directory the header resides in) visible to all libraries which link "_cpp-dns".